datalake/src/main/java/com/sequenceiq/datalake/flow/chain/DatalakeResizeRecoveryFlowEventChainFactory.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.datalake.flow.chain;
+
+import static com.sequenceiq.datalake.flow.delete.SdxDeleteEvent.SDX_DELETE_EVENT;
+import static com.sequenceiq.datalake.flow.detach.SdxDetachEvent.SDX_DETACH_RECOVERY_EVENT;
+import static com.sequenceiq.datalake.flow.detach.event.DatalakeResizeRecoveryFlowChainStartEvent.SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT;
+import static com.sequenceiq.datalake.flow.start.SdxStartEvent.SDX_START_EVENT;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.flow.delete.event.SdxDeleteStartEvent;
+import com.sequenceiq.datalake.flow.detach.event.DatalakeResizeRecoveryFlowChainStartEvent;
+import com.sequenceiq.datalake.flow.detach.event.SdxStartDetachRecoveryEvent;
+import com.sequenceiq.datalake.flow.start.event.SdxStartStartEvent;
+import com.sequenceiq.flow.core.chain.FlowEventChainFactory;
+import com.sequenceiq.flow.core.chain.config.FlowTriggerEventQueue;
+
+@Component
+public class DatalakeResizeRecoveryFlowEventChainFactory implements FlowEventChainFactory<DatalakeResizeRecoveryFlowChainStartEvent> {
+    @Override
+    public String initEvent() {
+        return SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT;
+    }
+
+    @Override
+    public FlowTriggerEventQueue createFlowTriggerEventQueue(DatalakeResizeRecoveryFlowChainStartEvent event) {
```

Comment:
could you add logs and unit test?

Reply:
What tests? I already have a test which verifies that the correct events are added to the queue, is there something else you want here?

Reply:
Added logs though.

Reply:
Added a lot more testing recently, please take a look.